PR to port resize_bilinear to TFLM broke the Xtensa build

https://github.com/tensorflow/tensorflow/pull/43426 broke the Xtensa build:

```
make -f tensorflow/lite/micro/tools/make/Makefile TARGET=xtensa OPTIMIZED_KERNEL_DIR=xtensa TARGET_ARCH=vision_p6 XTENSA_CORE=P6_200528 micro_allocator_test -j8
```

passes if we revert the PR and fails on tip of tree with:
```
xt-clang++ -std=c++11 -fno-exceptions -fno-threadsafe-statics -fno-unwind-tables -ffunction-sections -fdata-sections -fmessage-length=0 -DTF_LITE_STATIC_MEMORY -DTF_LITE_DISABLE_X86_NEON -O3 -Werror -Wsign-compare -Wdouble-promotion -Wshadow -Wunused-variable -Wmissing-field-initializers -Wunused-function -Wswitch -Wvla -Wall -Wextra -Wstrict-aliasing -Wno-unused-parameter -DXTENSA -DXTENSA -DTF_LITE_MCU_DEBUG_LOG -DTF_LITE_USE_CTIME --xtensa-core=P6_200528 -mcoproc -DMAX_RFFT_PWR=9 -DMIN_RFFT_PWR=MAX_RFFT_PWR -DVISION_P6 -I. -Itensorflow/lite/micro/tools/make/downloads/gemmlowp -Itensorflow/lite/micro/tools/make/downloads/flatbuffers/include -Itensorflow/lite/micro/tools/make/downloads/ruy -o tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/bin/micro_allocator_test tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/obj/tensorflow/lite/micro/micro_allocator_test.o tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/obj/tensorflow/lite/micro/testing/test_conv_model.o tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/lib/libtensorflow-microlite.a -Wl,--fatal-warnings -Wl,--gc-sections -lm
tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/lib/libtensorflow-microlite.a(memory_helpers.o): In function `tflite::BytesRequiredForTensor(tflite::Tensor const&, unsigned int*, unsigned int*, tflite::ErrorReporter*)':
memory_helpers.cc:(.text._ZN6tflite22BytesRequiredForTensorERKNS_6TensorEPjS3_PNS_13ErrorReporterE+0x13f): dangerous relocation: call8: call target out of range: _Assert
tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/lib/libtensorflow-microlite.a(micro_allocator.o): In function `tflite::internal::InitializeTfLiteTensorFromFlatbuffer(tflite::SimpleMemoryAllocator*, bool, tflite::Tensor const&, flatbuffers::Vector<flatbuffers::Offset<tflite::Buffer> > const*, tflite::ErrorReporter*, TfLiteTensor*)':
micro_allocator.cc:(.text._ZN6tflite8internal36InitializeTfLiteTensorFromFlatbufferEPNS_21SimpleMemoryAllocatorEbRKNS_6TensorEPKN11flatbuffers6VectorINS6_6OffsetINS_6BufferEEEEEPNS_13ErrorReporterEP12TfLiteTensor+0x47d): dangerous relocation: call8: call target out of range: _Assert
tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/lib/libtensorflow-microlite.a(micro_allocator.o): In function `tflite::internal::InitializeTfLiteEvalTensorFromFlatbuffer(tflite::SimpleMemoryAllocator*, tflite::Tensor const&, flatbuffers::Vector<flatbuffers::Offset<tflite::Buffer> > const*, tflite::ErrorReporter*, TfLiteEvalTensor*)':
micro_allocator.cc:(.text._ZN6tflite8internal40InitializeTfLiteEvalTensorFromFlatbufferEPNS_21SimpleMemoryAllocatorERKNS_6TensorEPKN11flatbuffers6VectorINS6_6OffsetINS_6BufferEEEEEPNS_13ErrorReporterEP16TfLiteEvalTensor+0x140): dangerous relocation: call8: call target out of range: _Assert
/home/advaitjain/xtensa/XtDevTools/install/tools/RI-2020.4-linux/XtensaTools/bin/xt-ld: error: 3 warnings, treating warnings as errors
clang-6.0: error: Xtensa-ld command failed with exit code 1 (use -v to see invocation)
make: *** [tensorflow/lite/micro/tools/make/Makefile:788: tensorflow/lite/micro/tools/make/gen/xtensa_vision_p6_default/bin/micro_allocator_test] Error 1
```

Couple of things to note:
 * The PR author (@patriklaurell) does not have any way to reproduce the errors, and we do not currently have any CI for the Xtensa toolchain.
 * Only one of 3 xtensa builds fails so there is a distinct possibility that the issue here is some subtlety with the Xtensa toolchain rather than the PR itself. However, the code that the Xtensa linker is complaining about it untouched by the PR that causes the error, so something odd is happening.
 * The micro_allocator_test is currently disabled for bluepill and stm32f4 (the two platforms that test for dynamic memory allocation etc.)
 * PR #43426 was created right around the time where TFLM had new guidelines for sending smaller PRs as part of OP porting and it was treated as an exception to the rule (which meant the PR is quite large and touches a lot of files).

Next steps:
 * https://github.com/tensorflow/tensorflow/pull/48515 reverts #43426 so that the Xtensa build is green again.
 * We should break up #43426 into smaller PRs following the newer guidelines for OP porting.
 * We should enable the micro_allocator_test on bluepill sooner rather than later

Tagging some of the folks that have a stake in this: @nyadla-sys @patriklaurell @freddan80 @petewarden

--- a/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc
+++ b/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc
@@ -39,7 +39,8 @@ PLATFORM_FLAGS = \
   -mcoproc \
   -DMAX_RFFT_PWR=9 \
   -DMIN_RFFT_PWR=MAX_RFFT_PWR \
-  $(TARGET_ARCH_DEFINES)
+  $(TARGET_ARCH_DEFINES) \
+  -mlongcalls
 
 ifeq ($(BUILD_TYPE), release)
   PLATFORM_FLAGS += -Wno-unused-private-field

